src/visualization/types/SimpleTable/PagedTable.tsx

```diff
@@ -292,7 +292,7 @@ const PagedTable: FC<Props> = ({result, properties}) => {
     if (maxSize) {
       setTotalPages(Math.ceil((result?.table?.length ?? 0) / maxSize))
     }
-  }, [maxSize]) // eslint-disable-line react-hooks/exhaustive-deps
+  }, [maxSize, result]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
This change fixes an issue where changing data buckets does not change the total number of pages in the pagination bar. Originally, this hook depended on the `maxSize` changing, but when you submit a new query with either the same bucket or a different one, the actual `maxSize` does not change, so the pagination bar doesn't update.